tests/tests_pytorch/serve/test_servable_module_validator.py

```diff
@@ -0,0 +1,41 @@
+from typing import Dict
+
+import torch
+
+from pytorch_lightning import seed_everything, Trainer
+from pytorch_lightning.demos.boring_classes import BoringModel
+from pytorch_lightning.serve.servable_module_validator import ServableModule, ServableModuleValidator
+
+
+class ServableBoringModel(BoringModel, ServableModule):
```

Comment:
I like this, but with this separate mixin, I strongly think that the validation/sanity checking shouldn't happen as part of a callback (and thereby inside the trainer). I think training and serving/deployment are completely different concerns and should also be treated that way.

Reply:
Again, I strongly disagree and I think this common sentiment among researchers is the reason why model mobility is so hard to do.

In my view, the model is the vessel that should implement all contracts from data processing to serving.

In PyTorch Lightning, the Trainer is the entry point. If I care about serving, then the training should validate my model is even worth training.